src/main_functions.c

```diff
@@ -0,0 +1,530 @@
+
+
+    parse_string(line, dict, underline, misspelled); //identify misspelled words and add to bad_word, 
+    //add to underline function 
+    
+    printf("Current line number is %d: \n", linenumber);
+    printf("%s", line_copy);
+    printf("\n");
+    printf("%s", underline);
+    printf("\n\n");
+
+    // printf("Misspelled words in this sentence are: ");
+
+    // int i = 0;
+    // while(badwords[i] != NULL) {
+    // 	printf("%d: %s ",i+1, badwords[i]);
+    // 	i++;
+    // }
+    // printf("\n");
+    // int number1;
+    // printf("Enter the number value of the word you would like to change: ");
+    // scanf("%d", &number1);
+
+    char *suggestions[max_no_suggestions]; //generates empty array where suggestions will be filled
+ 
+    int i = 0;
+
+    //replacing words according to user suggestions
+      while (misspelled[i] != NULL) {
+    	int success = generate_suggestions(misspelled[i], dict, suggestions);
+
+    	if(success != -1) {
+    	printf("Possible replacements for word %s are: ", misspelled[i]);
+    	printf("0: Delete Word. ");
+    	printf("1: No replacement. ");
+        int j = 0;
+    	for ( ; j < max_no_suggestions; j++) {
+    		printf("%d : %s ", j+2, suggestions[j]);
+
+    	}  	
+
+       }
+       printf("\n\n");
+
+    
+    //gets replacement choice from user
+    int number2;
+    printf("Enter the number of the replacement: ");
+
+    int check = scanf("%d", &number2);
+    assert(check >= 0);
+
+    if (number2 == 0) {
+    	printf("Deleting %s.", misspelled[i]);
+    	correct_line(line_copy, misspelled[i], "");
+    	printf("New sentence is: \n\n");
+     	printf("%s\n\n", line_copy);
+     	//printf("%s\n", badwords[i+1]);
+     	printf("%s\n", underline_misspelled_sentence(misspelled, line_copy, i+1));
+
+
+    }
+
+    else if (number2 == 1) {
+    	printf("No changes made to %s. \n\n", misspelled[i]);
+
+    }
+
+    else if (number2 > (max_no_suggestions+1)) {
+    	shell_error("Please enter a valid number");
+    }
+
+    else if (number2 != 1 || number2 != 0) { //1 if no replacement needed, 0 if word deleted
+    	printf("Replacing %s with %s \n", misspelled[i], suggestions[number2-2]);
+    	correct_line(line_copy, misspelled[i], suggestions[number2-2]); //modifies line function
+    	printf("New sentence is: \n");
+     	printf("%s\n\n", line_copy);
+     	printf("%s\n", underline_misspelled_sentence(misspelled, line_copy, i+1));
+    }
+
+
+	i++;	// added loop changer
+  }
+
+	return line_copy;
+	// need a way for string to (a) preserve punctuations and (b) 
+	// @Sarika this would be where the program needs replace_word, ignore_word, alternate_spelling
+}
+
+
+char** interactive_mode(char* filename, dict_t* dict, int* quit) //will pass in dictionary later
+{
+
+	size_t max_terminal_line_length = 500; // estiate
+	char** lines;
+	lines = parse_file(filename);
+
+	// step through phases
+	int i=0;
+	while (lines[i] != NULL) {	// potential error - one empty line in the middle of two full?	
```

Comment:
Have we tested for this error? This could cause the program to end early.

Reply:
lines will be parse_file; if there is an empty line, it will detect \n. error testing done in test_parser.